bot/modules/nostr/index.js

```diff
@@ -0,0 +1,71 @@
+require('websocket-polyfill');
+const Nostr = require('nostr-tools');
+const logger = require('../../../logger');
+const { userMiddleware } = require('../../middleware/user');
+const { orderCreated } = require('./events');
+const Config = require('./config');
+
+const pool = new Nostr.SimplePool();
+const relays = (env => {
+  if (!env.RELAYS)
+    return ['wss://nostr-pub.wellorder.net', 'wss://relay.damus.io'];
+  return env.RELAYS.split(',');
+})(process.env);
+relays.map(relay => pool.ensureRelay(relay));
+
+exports.addRelay = relay => {
+  relays.push(relay);
+  relays.map(relay => pool.ensureRelay(relay));
+};
+exports.getRelays = () => relays;
+
+exports.configure = bot => {
+  bot.command('nostr', async ctx => {
+    try {
+      const publicKey = Config.getPublicKey();
+      if (!publicKey) return;
+      const info = {
+        publicKey,
+        npub: Nostr.nip19.npubEncode(publicKey),
+        relays: relays.map(r => `<code>${r}</code>`).join('\n'),
+      };
+      const str = ctx.i18n.t('nostr_info', info);
+      await ctx.reply(str, { parse_mode: 'HTML' });
+    } catch (err) {
+      logger.error(err);
+      return ctx.reply('Error');
+    }
+  });
+  bot.command('setnpub', userMiddleware, async ctx => {
+    try {
+      const [, npub] = ctx.message.text.trim().split(' ');
+      const { type, data } = Nostr.nip19.decode(npub);
```

Comment:
if the user execute it without arguments it fires here an error trying to decode an empty npub, it should return how to use this command, something like

`/setnpub npub1...`